src/tutorials/data/queries.md

```diff
@@ -0,0 +1,365 @@
+
+⚠️ This implies that a query performed after many writes can take a certain time to complete, as it will update the index before returning any result. Likewise, the first query after an index creation will include the index build. Hence, developers should keep this behaviour in mind when designing applications and typically might need to handle query latency. 
+
+ℹ️ Starting from CouchDB 3.0.0, a [background indexing](https://docs.couchdb.org/en/3.0.0/config/indexbuilds.html) is implemented that aims to avoid this latency by forcing the index to update in background after writes operations.
+
+### Index with several fields
+
+Here is how to declare a query asking for todos having a “work” category OR a creation date starting from 2019:
+
+```javascript
+const queryDef = client
+  .find("io.cozy.todos")
+  .where({
+    "$or": [
+      {
+        "created_at": {
+          "$gt": "2019-01-01"
+      },
+      {
+        "category": "work"
+      }
+    ]
+  })
+  .indexFields(["category", "created_at"])
+```
+
+⚠️ The order in the `indexFields` array can be very important for performances, especially when there are both range and equality operators: you must always index the field involved in the equality operator first. To understand this, one needs to understand some advanced concepts about CouchDB indexes, explained in [this section](#indexes-performances-and-design).
+
+
+## Sort data with Mango
+
+It is possible to express the order of the documents returned by a query, by using the `sort`  array. 
+
+The sort is always done on a document field, by ascending (`asc`) or descending (`desc`) order, ascending being the default.  One can specify several fields to sort,  by following this structure for each field:
+
+`{"fieldName": "asc"|"desc"}`
+
+For instance:
+
+`[{"category": "asc"}, {"created_at": "asc"}]`
+
+Here, the returned documents will be first sorted by category, and then by creation date.
+
+⚠️ It is not possible to mix the sort order on different fields, i.e. all the fields involved in the sort must be either `asc`, either `desc` but not a mix of both.
+
+See [this section](#sort-in-couchdb) for more info about the sort order on data types.
+
+### Sort with cozy-client
+
+With cozy-client, a sort is expressed through the `sortBy` method, that takes into input an array containing the fields to sort and the direction. For instance:
+
+```javascript
+const queryDef = client
+  .find("io.cozy.todos")
+  .where({
+    "category": "sport",
+  })
+  .indexFields(["category", "created_at"])
+  .sortBy[{"category": "asc"}, {"created_at": "asc"}]
+```
+
+⚠️ Any field involved in the `sortBy` must be indexed: otherwise, CouchDB would be forced to scan all the documents in the database, which is not a good for scalability.
+
+⚠️ Inversely, when using a sort, any indexed field must appear either in the `where` selector or the `sortBy`.  
```

Comment:
Not `or`. It is not OK to index on `category, created_at`, have a selector on `category` and sort on `created_at` (CouchDB is a bit stupid).

Reply:
I just checked, and actually, when using a `category, created_at` index, it is necessary to either have a selector on both fields, either have a selector on `category` and a sort on `created_at`. Otherwise, the index is not used at all.